src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserRegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int LOGIN_MIN_LENGTH = 6;
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private static RegistrationService registrationServiceImpl;
+    private static User validUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationServiceImpl = new RegistrationServiceImpl();
+        validUser = User.of("valid_login", "valid_password", 32);
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void validUserCase() {
+        User expected = validUser;
+        User actual = registrationServiceImpl.register(validUser);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void validUserCase_withMinLengthLogin() {
+        User newValidUser = User.of(validUser.getLogin().substring(0, LOGIN_MIN_LENGTH),
+                validUser.getPassword(), validUser.getAge());
+        User expected = newValidUser;
+        User actual = registrationServiceImpl.register(newValidUser);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void validUserCase_withMinAge() {
+        User newValidUser = User.of(validUser.getLogin(), validUser.getPassword(), MIN_AGE);
+        User expected = newValidUser;
+        User actual = registrationServiceImpl.register(newValidUser);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void validUserCase_withMinLengthPassword() {
+        User newValidUser = User.of(validUser.getLogin(),
+                validUser.getPassword().substring(0, PASSWORD_MIN_LENGTH), validUser.getAge());
+        User expected = newValidUser;
+        User actual = registrationServiceImpl.register(newValidUser);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void userWithExistingLogin() {
```

Comment:
pease try to keep naming convention, if you add validUserCase before valid use cases please add invalidUserCase before invalid one